python/ray/util/collective/collective_group/nccl_collective_group.py

```diff
@@ -481,10 +543,19 @@ def _get_nccl_p2p_communicator(self, comm_key, my_gpu_idx, peer_rank,
         # create the p2p communicators
         with nccl_util.Device(my_gpu_idx):
             comm = nccl_util.create_nccl_communicator(2, nccl_uid, my_p2p_rank)
-            stream = cupy.cuda.Stream.null
-            # Stream(non_blocking=True)
+            #TODO(Fu): get the stream from pool
+            if _MULTI_STREAM:
+                pool: StreamPool = self._dev_pool_map.get(comm_key, StreamPool(self.POOL_SIZE))
```

Comment:
same here. You don't have code to manage the stream pool for each device at all. The current code will create stream pools every time when you try to query. This pool makes no sense.